### ISSUE TITLE
Can't sign in to personal account after signing into business account

### ISSUE THREAD

**Issue by AbeIonic** [dropbox/dropbox-sdk-obj-c#62]
After a user signs in with their business account, they can no longer attempt to add an additional account. It will skip the sign in page of the dropbox webview and request that they allow our app to access their files. This is a problem because we need to allow multiple accounts, business and personal. 

Even if we delete the access token for the business account, users can no longer sign into a different account. This issue doesn't occur when adding multiple personal accounts.

**Comment by greg-db** [comment]
Hi @AbeIonic can you share the relevant code snippet(s) and some screenshots or a screencast showing the issue so we can make sure we reproduce exactly what you're describing? Thanks in advance!

**Comment by AbeIonic** [comment]
Here are screenshots of our work flow for adding a Dropbox account to our app. First we add an enterprise account with SSO enabled, it redirects us to the SSO sign in, then requests permission from the user. After successfully adding that account we try to add another and it skips the sign in steps and jumps straight to the permission request for the previously added account.

![1](https://cloud.githubusercontent.com/assets/23510626/20400707/ce272eb0-acc3-11e6-8bd9-9c5b4fa067b7.jpg)
![2](https://cloud.githubusercontent.com/assets/23510626/20400703/ce23c824-acc3-11e6-9ed3-d7acf65c147e.jpg)
![3](https://cloud.githubusercontent.com/assets/23510626/20400704/ce242dd2-acc3-11e6-8761-ed1d316b2050.jpg)
![4](https://cloud.githubusercontent.com/assets/23510626/20400706/ce264766-acc3-11e6-986e-f18abde24bf7.jpg)
![5](https://cloud.githubusercontent.com/assets/23510626/20400708/ce27a688-acc3-11e6-8fd2-1f69a435b257.jpg)
![6](https://cloud.githubusercontent.com/assets/23510626/20400705/ce25eb90-acc3-11e6-8a1b-21f44c2a08bb.jpg)
![7](https://cloud.githubusercontent.com/assets/23510626/20400709/ce2e7c60-acc3-11e6-9a55-c03ed2da61ba.jpg)
![8](https://cloud.githubusercontent.com/assets/23510626/20400710/ce3301f4-acc3-11e6-8868-c6948bde9025.jpg)

We setup with:
```
DBMobileOAuthManager *oAuthManager = [[DBMobileOAuthManager alloc] initWithAppKey:kSRDropboxAppKey];
    [DBMobileOAuthManager sharedOAuthManager:oAuthManager];
```

We dispatch the connect account call with:
```
DBMobileSharedApplication *sharedMobileApplication = [[DBMobileSharedApplication alloc] init:[UIApplication sharedApplication]
                                                                                          controller:viewController
                                                                                           openURL:^(NSURL *_Nonnull url) {
                                                                                                 [[UIApplication sharedApplication] openURL:url];
                                                                                             }];
        [[DBOAuthManager sharedOAuthManager] authorizeFromSharedApplication:sharedMobileApplication browserAuth:NO];
```

This problem persists until the app is reinstalled.

**Comment by greg-db** [comment]
Thanks! To clarify though, are you _only_ seeing this with accounts using SSO, or is it any Business account?

**Comment by greg-db** [comment]
After further investigation, this does seem to be specific to using the SSO flow. We're looking into it.

**Comment by AbeIonic** [comment]
Has there been any progress on this or an ETA by chance? We really appreciate your support on this!

**Comment by greg-db** [comment]
I don't have an update on this yet unfortunately. I'll follow up here once I do.

**Comment by AbeIonic** [comment]
Can you give us any update on this? Is there a timeframe when you think you'll be able to tackle it? It's a serious issue in our app.

**Comment by greg-db** [comment]
We have an understanding of the problem, but I can't offer a timeline for a fix right now. 

If it helps, as an (albeit non-ideal) workaround for affected users though, they should be able to revoke the relevant web session via https://www.dropbox.com/account/security .

**Comment by AbeIonic** [comment]
Alright, thank you. We look forward to the fix when you guys get the chance :)

**Comment by scobbe** [comment]
@AbeIonic: Sorry you're running into issues here. I put out a fix in the latest version of the SDK that might fix this issue for you. Would you try pulling the (just-released) latest version of the SDK (2.0.0) and see if you still experience these issues? Thanks for your patience!

**Comment by AbeIonic** [comment]
This has solved our problem. Thank you very much!
